File: 2015/day02-1.py
f = open("day02-puzzle-input.txt", "r")

total = 0

for ligne in f:
    split = ligne.split('x')
    l = int(split[0])
    w = int(split[1])
    h = int(split[2])

    minimum = min(l*w, w*h, h*l)
    print((l, w, h, minimum))

    total += 2*l*w + 2*w*h + 2*h*l + minimum

print(total)

File: 2021/day02-1.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	contenu, _ := os.ReadFile("day02-puzzle-input.txt")

	tableau := strings.Split(string(contenu), "\n")
	horizontal := 0
	descente := 0

	for i := range tableau {
		commande := strings.Split(tableau[i], " ")
		valeur, _ := strconv.Atoi(commande[1])
		switch commande[0] {
		case "up":
			descente -= valeur
		case "down":
			descente += valeur
		case "forward":
			horizontal += valeur
		}
	}

	fmt.Println(horizontal * descente)
}
